src/client/fontengine.cpp

```diff
@@ -162,6 +163,15 @@ void FontEngine::readSettings()
 	refresh();
 }
 
+void FontEngine::handleReload()
+{
+	if (!m_needs_reload)
+		return;
+
+	m_needs_reload = false;
+	readSettings();
```

Comment:
Couldn't you call `readSettings` within `fontSettingChanged`? In that case you wouldn't need a separate variable and update call.